api/v1alpha2/azurecluster_conversion.go

```diff
@@ -57,6 +57,17 @@ func (src *AzureCluster) ConvertTo(dstRaw conversion.Hub) error { // nolint
 	}
 
 	dst.Status.FailureDomains = restored.Status.FailureDomains
+	//dst.Spec.NetworkSpec.Subnets = restored.Spec.NetworkSpec.Subnets
+
+	// for _, restoredSubnet := range restored.Spec.NetworkSpec.Subnets {
```

Comment:
@alexeldeib @devigned I'm struggling to figure what is the correct way to restore here... It's slightly different from `Files.ContentFrom` because there is no mutual exclusivity with another field like `Content` and `ContentFrom`, it's just a new optional field. The above passes the fuzzer unit test but I think it's incorrect because it might overwrite existing v1alpha2 fields that have changed. The below is my attempt to only restore `RouteTable` based on `Name` (assuming `Name` doesn't change that's arguably better than by index but it's not ideal), and that is failing the test.